internal/decomposer/decomposer.go

```diff
@@ -69,11 +69,18 @@ type Decomposer struct {
 }
 
 // Execute a Playbook
-func (decomposer *Decomposer) Execute(playbook cacao.Playbook) (*ExecutionDetails, error) {
+func (decomposer *Decomposer) Execute(playbook cacao.Playbook, execution_details_ch chan ExecutionDetails) (*ExecutionDetails, error) {
```

Comment:
The Exection details can be moved into two different functions and the rest of the execution can be doen in execute(..)